### PR TITLE
Add db->perform INSERT IGNORE

### PR DESCRIPTION
**Pull request by drbyte**
Previously only INSERT or UPDATE were allowed, with no way to IGNORE.

### CODE REVIEW

File: includes/classes/db/mysql/query_factory.php [hunk 1]
```diff
@@ -477,23 +477,25 @@ public function queryTime(): int
...
      */
-    public function perform(string $tableName, array $tableData, $performType = 'INSERT', string $whereCondition = '', $debug = false): void
+    public function perform(string $tableName, array $tableData, string $performType = 'INSERT', string $whereCondition = '', ?bool $debug = false): void
     {
         switch (strtolower($performType)) {
+            case 'insertignore':
+                $insertString = "INSERT IGNORE INTO " . $tableName . " (";
             case 'insert':
-                $insertString = "INSERT INTO " . $tableName . " (";
+                $insertString = $insertString ?? "INSERT INTO " . $tableName . " (";
```
Comment (lat9) [inline includes/classes/db/mysql/query_factory.php:486]: For dis-ambiguity, should the `"INSERT INTO " . $tableName . " ("` portion be wrapped in parentheses?

File: includes/classes/db/mysql/query_factory.php [hunk 2]
```diff
@@ -503,13 +505,15 @@ public function perform(string $tableName, array $tableData, $performType = 'INS
 
                 break;
 
+            case 'updateignore':
+                $updateString = 'UPDATE IGNORE ' . $tableName . ' SET ';
             case 'update':
-                $updateString = 'UPDATE ' . $tableName . ' SET ';
+                $updateString = $updateString ?? 'UPDATE ' . $tableName . ' SET ';
```
Comment (lat9) [inline includes/classes/db/mysql/query_factory.php:511]: Ditto for the `UPDATE` concatenation.

**Review by lat9 — COMMENTED** [review]
Just comments to prevent potential ambiguity issues in the future.

**Comment by scottcwilson** [comment]
Thanks for doing this - it will make the updates log less!

File: includes/classes/db/mysql/query_factory.php (hunk 1 above)
Comment (drbyte) [reply includes/classes/db/mysql/query_factory.php]: Done. Albeit differently.

**Pull request merged**